### PR TITLE
#3423 Robot Framework JSON output

### PR DESCRIPTION
**Pull request by Lemonlemmings**

### CODE REVIEW

File: src/robot/conf/settings.py [hunk 1]
```diff
@@ -203,7 +204,7 @@ def __getitem__(self, name):
     def _get_output_file(self, option):
         """Returns path of the requested output file and creates needed dirs.
 
-        `option` can be 'Output', 'Log', 'Report', 'XUnit' or 'DebugFile'.
+        `option` can be 'Output', 'Log', 'Report', 'XUnit', 'Junit', or 'DebugFile'.
```
Comment (laurentbristiel) [inline src/robot/conf/settings.py]: Junit => Json

File: src/robot/output/jsonlogger.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+            parent_item = self._current_suite
+        elif destination == 'test':
+            parent_item = self._current_test
+        else:
+            parent_item = self._item_stack[-1]
+
+        if 'kw' not in parent_item:
+            parent_item['kw'] = list()
+        parent_item['kw'].append(self._current_keyword)
+        # Mark the current keyword as being none
```
Comment (laurentbristiel) [inline src/robot/output/jsonlogger.py]: not sure these 2 comments have much added value. The code is clear enough.

File: src/robot/output/jsonlogger.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+            self._current_suite['tests'] = list()
+        self._current_suite['tests'].append(self._current_test)
+
+    def start_suite(self, suite):
+        # If there is an "open" suite, this will be placed inside its suites
+        if self._current_suite:
+            # Push this onto the stack
+            self._item_stack.append(self._current_suite)
+
+        # Default the current suite with t
```
Comment (laurentbristiel) [inline src/robot/output/jsonlogger.py]: with t ?

File: src/robot/conf/settings.py (hunk 1 above)
Comment (Lemonlemmings) [reply src/robot/conf/settings.py]: Good catch! Resolved.

File: src/robot/output/jsonlogger.py (hunk 2 above)
Comment (Lemonlemmings) [reply src/robot/output/jsonlogger.py]: I've removed the comments

File: src/robot/output/jsonlogger.py (hunk 3 above)
Comment (Lemonlemmings) [reply src/robot/output/jsonlogger.py]: Nice catch, resolved!

**Comment by pekkaklarck** [comment]
Thanks for your efforts with this PR @Lemonlemmings. Unfortunately I didn't have time to review it fully back when it was created and now there are so bad conflicts that I think it's best to just close it.

Good news is that there has been a lot of progress with JSON support otherwise:
- The execution model got JSON serialization support in RF 6.1 (#3902).
- The result model got it in RF 7.0 (#4847).
- The result model will be enhanced a bit in RF 7.2 (#5160).

The final missing part is supporting JSON as part of the execution. That enhancement is covered by #3423 and it's currently targeted for RF 7.2 as well. With all earlier work that shouldn't anymore be too big a task, but due to time constraints there's a risk it will be moved to RF 7.3.

If you are still interested to help with this enhancement, let me know and we can collaborate in getting the aforementioned final piece done. I also totally understand if you don't want to spend more time for this as the previous PR didn't get anywhere and you obviously may not be working with Robot Framework anymore either.

**Pull request closed without merging**